{"ygPerms":{"resourceCapabilityList":[{"resourceType":"GROUP","capabilities":[{"name":"READ"},{"name":"JOIN"}]},{"resourceType":"PHOTO","capabilities":[]},{"resourceType":"FILE","capabilities":[]},{"resourceType":"MEMBER","capabilities":[]},{"resourceType":"LINK","capabilities":[]},{"resourceType":"CALENDAR","capabilities":[]},{"resourceType":"DATABASE","capabilities":[]},{"resourceType":"POLL","capabilities":[]},{"resourceType":"MESSAGE","capabilities":[{"name":"READ"}]},{"resourceType":"PENDING_MESSAGE","capabilities":[]},{"resourceType":"ATTACHMENTS","capabilities":[{"name":"READ"}]},{"resourceType":"PHOTOMATIC_ALBUMS","capabilities":[]},{"resourceType":"MEMBERSHIP_TYPE","capabilities":[]},{"resourceType":"POST","capabilities":[{"name":"READ"}]},{"resourceType":"PIN","capabilities":[]}],"groupUrl":"groups.yahoo.com","intlCode":"us"},"comscore":"pageview_candidate","ygData":{"userId":137285340,"authorName":"Gordon Mohr","from":"Gordon Mohr &lt;gojomo@...&gt;","profile":"gojomo","replyTo":"LIST","senderId":"adBPWTamQLFV13_HSMcSdAIxfrAmYYmwBKkcIFPu7CQlsoM50ep1YjMMb9oWubt1lyr9R5LqidJLziT0goTLfWpkayRzwa4","spamInfo":{"isSpam":false,"reason":"12"},"subject":"Re: [archive-crawler] Exception when instantiating Processor (when launching crawl job with a custom processor)","postDate":"1212435834","msgId":5270,"canDelete":false,"contentTrasformed":false,"systemMessage":false,"headers":{"messageIdInHeader":"PDQ4NDQ0RDdBLjgwMTA4MDVAYXJjaGl2ZS5vcmc+","inReplyToHeader":"PGQ2MWM1NzMwMDgwNjAyMDgwMnA3MTE1NjRjOG4yZmVmOTJlMjJiZTVlMGYyQG1haWwuZ21haWwuY29tPg==","referencesHeader":"PGQ2MWM1NzMwMDgwNTE0MDcwM3A0OWUxYzFjZXAxMDIzNjAxZTdkYWFkOThlQG1haWwuZ21haWwuY29tPgkgPGFlNDc2N2QxMDgwNTE0MTcyOGs0MjVhNjUxYXUxODczMTU0ODExNzJiZWE4QG1haWwuZ21haWwuY29tPgkgPGQ2MWM1NzMwMDgwNTE1MDEyNGwxZWNiZDZiNXViMjY1YmQ0MzE4ODkzMDE2QG1haWwuZ21haWwuY29tPiA8ZDYxYzU3MzAwODA2MDIwODAycDcxMTU2NGM4bjJmZWY5MmUyMmJlNWUwZjJAbWFpbC5nbWFpbC5jb20+"},"prevInTopic":5264,"nextInTopic":0,"prevInTime":5269,"nextInTime":5271,"topicId":5205,"numMessagesInTopic":9,"msgSnippet":"I don t know what s happening, but parts of your description don t add up. In particular, all the concrete Processor classes standard with Heritrix have","rawEmail":"Return-Path: &lt;gojomo@...&gt;\r\nX-Sender: gojomo@...\r\nX-Apparently-To: archive-crawler@yahoogroups.com\r\nX-Received: (qmail 71947 invoked from network); 2 Jun 2008 19:43:50 -0000\r\nX-Received: from unknown (66.218.67.97)\n  by m36.grp.scd.yahoo.com with QMQP; 2 Jun 2008 19:43:50 -0000\r\nX-Received: from unknown (HELO relay02.pair.com) (209.68.5.16)\n  by mta18.grp.scd.yahoo.com with SMTP; 2 Jun 2008 19:43:49 -0000\r\nX-Received: (qmail 8443 invoked from network); 2 Jun 2008 19:43:48 -0000\r\nX-Received: from unknown (HELO ?192.168.1.15?) (unknown)\n  by unknown with SMTP; 2 Jun 2008 19:43:48 -0000\r\nX-pair-Authenticated: 67.180.197.118\r\nMessage-ID: &lt;48444D7A.8010805@...&gt;\r\nDate: Mon, 02 Jun 2008 12:43:54 -0700\r\nUser-Agent: Thunderbird 2.0.0.14 (Windows/20080421)\r\nMIME-Version: 1.0\r\nTo: archive-crawler@yahoogroups.com\r\nReferences: &lt;d61c57300805140703p49e1c1cep1023601e7daad98e@...&gt;\t &lt;ae4767d10805141728k425a651au187315481172bea8@...&gt;\t &lt;d61c57300805150124l1ecbd6b5ub265bd4318893016@...&gt; &lt;d61c57300806020802p711564c8n2fef92e22be5e0f2@...&gt;\r\nIn-Reply-To: &lt;d61c57300806020802p711564c8n2fef92e22be5e0f2@...&gt;\r\nContent-Type: text/plain; charset=UTF-8; format=flowed\r\nContent-Transfer-Encoding: 8bit\r\nX-eGroups-Msg-Info: 1:12:0:0:0\r\nFrom: Gordon Mohr &lt;gojomo@...&gt;\r\nSubject: Re: [archive-crawler] Exception when instantiating Processor (when\n launching crawl job with a custom processor)\r\nX-Yahoo-Group-Post: member; u=137285340; y=N5RSeP9vNKvLy82HTuf-I3zIxRry7DZggZ_ZInRjI0tQ\r\nX-Yahoo-Profile: gojomo\r\n\r\nI don&#39;t know what&#39;s happening, but parts of your description don&#39;t add up.\n\nIn particular, all the concrete Processor classes standard with Heritrix \nhave no-argument constructors, and don&#39;t trigger any problem. Indeed, a \nProcessor class needs to have a no-argument constructor to be instantiable.\n\nSo, removing such a constructor would also mean no instance of that \nProcessor is part of the launched configuration.\n\nPerhaps your custom Processor has settings Key&lt;&gt;s, but does not register \nthem with the settings system via the KeyManager.addKeys() \nstatic-initializer code you&#39;ll see at the bottom of other configurable \nclasses?\n\n- Gordon @ IA\n\nJean-Noël Rivasseau wrote:\n&gt; *bump*\n&gt; \n&gt; Still not understanding why this happens.\n&gt; \n&gt; On Thu, May 15, 2008 at 10:24 AM, Jean-Noël Rivasseau &lt;elvanor@...&gt; wrote:\n&gt;&gt; Hmm, no I am not debugging in Eclipse or doing anything fancy there...\n&gt;&gt;\n&gt;&gt; On Thu, May 15, 2008 at 2:28 AM, Ryan Smith &lt;ryan.justin.smith@...&gt; wrote:\n&gt;&gt;&gt; Not sure if this is going to help you, but i was getting a similar error\n&gt;&gt;&gt; when trying to debug in eclipse, and i jsut laoded all the binary 2rd party\n&gt;&gt;&gt; lib jars on the runtime classpath and then i could use the webui fine.\n&gt;&gt;&gt;\n&gt;&gt;&gt; I was having this error happen to me when i tried to edit a global Sheet\n&gt;&gt;&gt; config.\n&gt;&gt;&gt;\n&gt;&gt;&gt; On Wed, May 14, 2008 at 10:03 AM, Jean-Noël Rivasseau &lt;elvanor@...&gt;\n&gt;&gt;&gt; wrote:\n&gt;&gt;&gt;&gt; Hello, I am very new to Heritrix and am using 2.0.0. When I launch a\n&gt;&gt;&gt;&gt; job I get the following exception:\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; I noticed this happens whenever I have a constructor defined in my\n&gt;&gt;&gt;&gt; custom Processor class that extends org.archive.modules.Processor.\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; This happens even if the constructor code is totally empty. If I\n&gt;&gt;&gt;&gt; remove the constructor everything works. However having a constructor\n&gt;&gt;&gt;&gt; to perform initialization would be nice...\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; Am I doing something wrong? Any help appreciated.\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; Thanks,\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; Jean-Noel\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; javax.management.ReflectionException\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; null\n&gt;&gt;&gt;&gt;\n&gt;&gt;&gt;&gt; javax.management.ReflectionException\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.archive.settings.jmx.LoggingDynamicMBean.dealWithException(LoggingDynamicMBean.java:77)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.archive.settings.jmx.LoggingDynamicMBean.invoke(LoggingDynamicMBean.java:160)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.invoke(DefaultMBeanServerInterceptor.java:836)\n&gt;&gt;&gt;&gt; at com.sun.jmx.mbeanserver.JmxMBeanServer.invoke(JmxMBeanServer.java:761)\n&gt;&gt;&gt;&gt; at org.archive.openmbeans.annotations.BeanProxy.invoke(BeanProxy.java:87)\n&gt;&gt;&gt;&gt; at $Proxy9.launchJob(Unknown Source)\n&gt;&gt;&gt;&gt; at org.archive.crawler.webui.CrawlerArea.launch(CrawlerArea.java:155)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; jsp.crawler_005farea.do_005flaunch_jsp._jspService(jsp.crawler_005farea.do_005flaunch_jsp:45)\n&gt;&gt;&gt;&gt; at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:97)\n&gt;&gt;&gt;&gt; at javax.servlet.http.HttpServlet.service(HttpServlet.java:809)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:459)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1054)\n&gt;&gt;&gt;&gt; at org.archive.crawler.webui.AuthFilter.doFilter(AuthFilter.java:79)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1045)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:358)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:231)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:629)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:453)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.handler.HandlerList.handle(HandlerList.java:49)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:141)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.Server.handle(Server.java:303)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:452)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:721)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:509)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:209)\n&gt;&gt;&gt;&gt; at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:349)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:217)\n&gt;&gt;&gt;&gt; at\n&gt;&gt;&gt;&gt; org.mortbay.thread.BoundedThreadPool$PoolThread.run(BoundedThreadPool.java:475)\n&gt;&gt;&gt;&gt; Caused by: java.lang.IllegalStateException\n&gt;&gt;&gt;&gt; ... 28 more\n&gt;&gt;&gt;\n&gt; \n&gt; ------------------------------------\n&gt; \n&gt; Yahoo! Groups Links\n&gt; \n&gt; \n&gt; \n\n"}}